providers/freebsd/memory_freebsd.go

```diff
@@ -0,0 +1,111 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build freebsd,cgo
+
+package freebsd
+
+// #cgo LDFLAGS: -lkvm
+//#include <sys/cdefs.h>
+//#include <sys/types.h>
+//#include <sys/sysctl.h>
+
+//#include <paths.h>
+//#include <kvm.h>
+//#include <stdlib.h>
+import "C"
+
+import (
+	"github.com/pkg/errors"
+	"syscall"
+	"unsafe"
+)
+
+const hwPhysmemMIB = "hw.physmem"
+const hwPagesizeMIB = "hw.pagesize"
+const vmVmtotalMIB = "vm.vmtotal"
+const vmSwapmaxpagesMIB = "vm.swap_maxpages"
+const vfsNumfreebuffersMIB = "vfs.numfreebuffers"
+const devNull = "/dev/null"
+const kvmOpen = "kvm_open"
+
+func PageSize() (uint32, error) {
+	var pageSize uint32
+	if err := sysctlByName(hwPagesizeMIB, &pageSize); err != nil {
+		return 0, errors.Wrap(err, "failed to get hw.pagesize")
+	}
+
+	return pageSize, nil
+}
+
+func SwapMaxPages() (uint32, error) {
+	var maxPages uint32
+	if err := sysctlByName(hwPhysmemMIB, &maxPages); err != nil {
```

Comment:
Should be `vmSwapmaxpagesMIB`

Reply:
For the record, it's `vm.swap_maxpages` OID.